vyper/builtin_functions/functions.py

```diff
@@ -178,21 +194,40 @@ class Convert(_BuiltinFunction):
     _id = "convert"
 
     def fetch_call_return(self, node):
+        self.infer_arg_types(node)
+        target_type = get_type_from_annotation(node.args[1], DataLocation.MEMORY)
+
+        # note: more type conversion validation happens in convert.py
+        return target_type
+
+    def infer_arg_types(self, node):
         validate_call_args(node, 2)
+
         target_type = get_type_from_annotation(node.args[1], DataLocation.MEMORY)
-        validate_expected_type(node.args[0], ValueTypeDefinition())
+        value_types = get_possible_types_from_node(node.args[0])
+        if len(value_types) == 0:
+            raise CompilerPanic("No possible type for value", node)
+
+        if all([isinstance(v, IntegerAbstractType) for v in value_types]):
+            # Get the smallest (and unsigned if available) type for non-integer output types
```

Comment:
this code is somewhat hard to follow. can it be put into a helper function which better communicates what it does? (or better yet -- can an existing helper function be re-used?)

Reply:
Made a tweak to `get_possible_types_from_node` instead. Note to self: check on CI